modules/encrypt/classes/Kohana/Encrypt/Engine.php

```diff
@@ -1,78 +1,91 @@
 <?php
 
-abstract class Kohana_Encrypt_Engine {
+/**
+ * @category   Security
+ * @package    Kohana/Encrypt
+ * @author     Kohana Team
+ * @copyright  (c) 2007-2012 Kohana Team
+ * @copyright  (c) 2016-2018 Koseven Team
+ * @license    https://koseven.ga/LICENSE.md
+ */
+abstract class Kohana_Encrypt_Engine
+{
+    /**
+     * @var string name of the engine in configuration
+     */
+    const CONFIG_TYPE = 'type';
 
-	/**
-	 * @var string Encryption key
-	 */
-	protected $_key;
+    /**
+     * @var string name of the key in configuration
+     */
+    const CONFIG_KEY = 'key';
```

Comment:
Don't assign default value to eky...many function check for that and will therefore not work.
e.G: 

Let's go through the flow (Lets pretend we are instancing `default` the first time.
1. You instance Encrypt Engine with `Encrypt::instance()`

Inside class `Encrypt`:

2. It assigns configuration name is `default`
3. Loads Configuration `default` from config file `encrypt`
4. Checks if COSNTANT  `Kohana_Encrypt_Engine::CONFIG_KEY` is set

Inside Kohana_Encrypt_Engine::CONFIG_KEY, we can see, that it has now a default value, therefore it will always return true. 

Same happens instancing `Kohana_Encrypt_Engine` inside `constructor` it checks if variable is empty...it isn't even if key is not set.

Reply:
No, no, no.
Those constants simply represent index in the $config array.
So before it was
```
return [
   'key' => 'SomeSecretKey'
];
$this->_key = $config['key'];
```

And now it is:
```
const CONFIG_KEY = 'key';
$this->_key = $config[self::CONFIG_KEY];
```

Reply:
oh yes i see thanks!